### PR TITLE
Add tests for KubeletConfig

### PR DESCRIPTION
**Pull request by tahsinrahman**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
This pr adds tests for `KubeletConfig` using the component-base testing package introduced here https://github.com/kubernetes/kubernetes/pull/84688 . This is a part of increasing testing coverage for ComponentConfigs. 

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```
/wg component-standard
/cc @mtaufen @stealthybox

### CODE REVIEW

File: staging/src/k8s.io/component-base/config/testing/helpers.go [hunk 1]
```diff
@@ -70,6 +71,10 @@ func RunTestsOnYAMLData(t *testing.T, scheme *runtime.Scheme, tests []TestCase,
 			if needsUpdate {
 				const updateEnvVar = "UPDATE_COMPONENTCONFIG_FIXTURE_DATA"
 				if os.Getenv(updateEnvVar) == "true" {
+					if err := os.MkdirAll(filepath.Dir(rt.out), 0755); err != nil {
```
Comment (mtaufen) [inline staging/src/k8s.io/component-base/config/testing/helpers.go]: hm, why did this work before?

File: staging/src/k8s.io/kubelet/config/v1beta1/register_test.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+func TestComponentConfigSetup(t *testing.T) {
+	pkginfo := &componentconfigtesting.ComponentConfigPackage{
+		ComponentName:      "kubelet",
+		GroupName:          GroupName,
+		SchemeGroupVersion: SchemeGroupVersion,
+		AddToScheme:        AddToScheme,
+	}
+
+	if err := componentconfigtesting.VerifyExternalTypePackage(pkginfo); err != nil {
+		t.Errorf("failed TestComponentConfigSetup: %v", err)
```
Comment (mtaufen) [inline staging/src/k8s.io/kubelet/config/v1beta1/register_test.go]: Don't need the test function name in the error message, the test framework will take care of that for us.

File: staging/src/k8s.io/kubelet/go.sum [hunk 3]
```diff
@@ -122,12 +194,15 @@ gopkg.in/yaml.v2 v2.2.2/go.mod h1:hI93XBmqTisBFMUTm0b8Fm+jr3Dg1NNxqwp+5A1VGuI=
 gopkg.in/yaml.v2 v2.2.4 h1:/eiJrUcujPVeJ3xlSWaiNi3uSVmDGBK1pDHUHAnao1I=
 gopkg.in/yaml.v2 v2.2.4/go.mod h1:hI93XBmqTisBFMUTm0b8Fm+jr3Dg1NNxqwp+5A1VGuI=
 honnef.co/go/tools v0.0.0-20190102054323-c2f93a96b099/go.mod h1:rf3lG4BRIbNafJWhAfAdb/ePZxsR/4RtNHQocxwk9r4=
+honnef.co/go/tools v0.0.0-20190106161140-3f1c8253044a/go.mod h1:rf3lG4BRIbNafJWhAfAdb/ePZxsR/4RtNHQocxwk9r4=
 honnef.co/go/tools v0.0.0-20190523083050-ea95bdfd59fc/go.mod h1:rf3lG4BRIbNafJWhAfAdb/ePZxsR/4RtNHQocxwk9r4=
 k8s.io/gengo v0.0.0-20190128074634-0689ccc1d7d6/go.mod h1:ezvh/TsK7cY6rbqRK0oQQ8IAqLxYwwyPxAX1Pzy0ii0=
 k8s.io/klog v0.0.0-20181102134211-b9b56d5dfc92/go.mod h1:Gq+BEi5rUBO/HRz0bTSXDUcqjScdoY3a9IHpCEIOOfk=
+k8s.io/klog v0.3.0/go.mod h1:Gq+BEi5rUBO/HRz0bTSXDUcqjScdoY3a9IHpCEIOOfk=
 k8s.io/klog v1.0.0 h1:Pt+yjF5aB1xDSVbau4VsWe+dQNzA0qv1LlXdC2dF6Q8=
 k8s.io/klog v1.0.0/go.mod h1:4Bi6QPql/J/LkTDqv7R/cd3hPo4k2DG6Ptcz060Ez5I=
```
Comment (mtaufen) [inline staging/src/k8s.io/kubelet/go.sum]: Why are there multiple versions of klog here (new 0.3 vs prev 1.0)? Should something be using the 1.0.0 version? Same thing seems to be happening for some other dependencies (multiple versions, and even not across a major version boundary).

@liggitt thoughts?

File: pkg/kubelet/apis/config/scheme/testdata/KubeletConfiguration/after/__internal.yaml [hunk 4]
```diff
@@ -0,0 +1,93 @@
+Address: ""
```
Comment (liggitt) [inline pkg/kubelet/apis/config/scheme/testdata/KubeletConfiguration/after/__internal.yaml]: don't test internal API types by serializing them, that is contrary to their intended purpose

Comment (liggitt) [reply pkg/kubelet/apis/config/scheme/testdata/KubeletConfiguration/after/__internal.yaml]: same goes for all the __internal.yaml files... they should not exist

File: staging/src/k8s.io/kubelet/go.sum (hunk 3 above)
Comment (liggitt) [reply staging/src/k8s.io/kubelet/go.sum]: The .sum file records checksums for all the versions of this dependency reported by various transitive dependencies, and is built by `go mod tidy`. It has no bearing on the version that is used. If the dependencies CI job is happy, don't worry about this file.

File: staging/src/k8s.io/component-base/config/testing/helpers.go (hunk 1 above)
Comment (tahsinrahman) [reply staging/src/k8s.io/component-base/config/testing/helpers.go]: the directory was already present on my machine, so it didn't complain :(

**Comment by liggitt** [comment]
can you squash commits to clean up the 0-line __internal files in the diff?

File: pkg/kubelet/apis/config/scheme/testdata/KubeletConfiguration/after/__internal.yaml (hunk 4 above)
Comment (mtaufen) [reply pkg/kubelet/apis/config/scheme/testdata/KubeletConfiguration/after/__internal.yaml]: That's a good point that I missed the first time around. We should also clean up the same in kube-proxy from the prior PR.

Comment (tahsinrahman) [reply pkg/kubelet/apis/config/scheme/testdata/KubeletConfiguration/after/__internal.yaml]: I created https://github.com/kubernetes/kubernetes/pull/86144 to address this issue! PTAL :)

**Comment by danwinship** [comment]
@tahsinrahman are you still working on this?

**Comment by tahsinrahman** [comment]
@danwinship I’ll work on this once https://github.com/kubernetes/kubernetes/pull/86144 is merged. Thanks!

File: staging/publishing/rules.yaml [hunk 5]
```diff
@@ -770,6 +770,10 @@ rules:
       branch: master
     - repository: api
       branch: master
+    - repository: client-go
```
Comment (mtaufen) [inline staging/publishing/rules.yaml]: This seems like a change that should be made in an independent PR?

Comment (tahsinrahman) [reply staging/publishing/rules.yaml]: Without it the test fails, so i added here https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/86034/pull-kubernetes-verify/1223171713915686913

Comment (mtaufen) [reply staging/publishing/rules.yaml]: It's possible this came from adding component-base as a dependency of the Kubelet apis subdir. @liggitt @sttts WDYT? Is this change ok?

Comment (liggitt) [reply staging/publishing/rules.yaml]: it is because this adds k8s.io/kubelet -> k8s.io/component-base -> k8s.io/client-go

I'll defer to @sttts on whether that layering makes sense, but I think it probably does

Comment (mtaufen) [reply staging/publishing/rules.yaml]: I think the layering makes sense, just wanted to make sure this wasn't an artifact of something else. Thanks!

Comment (sttts) [reply staging/publishing/rules.yaml]: +1

**Comment by tahsinrahman** [comment]
@mtaufen i've rebased the pr
